core/dnsserver/address.go

```diff
@@ -94,3 +101,43 @@ const (
 	TransportTLS  = "tls"
 	TransportGRPC = "grpc"
 )
+
+// Build a Validator that rise error if the bound addresses for listeners are overlapping
```

Comment:
This comment doesn't really describe what this struct does, but what you're planning to do with it, so this should be mostly moved to `registerAndCheck`, there are also a bunch of typos in it.

Reply:
ok. move and fix typos